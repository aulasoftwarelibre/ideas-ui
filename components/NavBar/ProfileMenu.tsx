import classnames from 'classnames';
import React from 'react';

import { profileLinks } from '../../config/links';
import { Session } from '../../model/user/auth';
import ProfileLink from './ProfileLink';

export interface Props {
  session: Session;
  hideMenu: boolean;
  onClick: () => void;
}

export const ProfileMenu: React.FunctionComponent<Props> = ({
  session,
  hideMenu,
  onClick,
}) => {
  return (
    <>
      {session && (
        <div
          className="absolute z-10 inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
          data-testid="profile-menu"
        >
          <div className="relative">
            <div>
              <button
                className="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
                id="user-menu"
                aria-label="User menu"
                aria-haspopup="true"
                onClick={onClick}
              >
                <img
                  className="h-8 w-8 rounded-full"
                  src={session?.user?.image ?? '/avatar.svg'}
                  alt={session?.user?.name}
                />
              </button>
            </div>
            <div className="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg">
              <div
                className={classnames('py-1 rounded-md bg-white shadow-xs', {
                  hidden: hideMenu,
                })}
                role="menu"
                aria-orientation="vertical"
                aria-labelledby="user-menu"
                data-testid="profile-menu-box"
              >
                {profileLinks.map((link) => (
                  <ProfileLink
                    key={link.path}
                    path={link.path}
                    label={link.label}
                  />
                ))}
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default ProfileMenu;
