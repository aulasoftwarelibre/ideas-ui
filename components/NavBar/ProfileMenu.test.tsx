import * as React from 'react';

import { mountMockRouter, renderMockRouter } from '../../utils/test-utils';
import ProfileMenu from './ProfileMenu';

describe("ProfileMenu", () => {
  it("should be rendered opened with session", () => {
    const { asFragment } = renderMockRouter(
      <ProfileMenu
        session={{ user: {} }}
        hideMenu={false}
        onClick={() => {}}
      />,
      {
        router: { pathname: "/" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("should be rendered closed with session", () => {
    const { asFragment } = renderMockRouter(
      <ProfileMenu session={{ user: {} }} hideMenu={true} onClick={() => {}} />,
      {
        router: { pathname: "/" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be rendered correctly without session", () => {
    const { asFragment } = renderMockRouter(
      <ProfileMenu session={null} hideMenu={false} onClick={() => {}} />,
      {
        router: { pathname: "/" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should open profile menu when click", () => {
    const mockOnClick = jest.fn();

    const wrapper = mountMockRouter(
      <ProfileMenu
        session={{ user: {} }}
        hideMenu={false}
        onClick={mockOnClick}
      />,
      {
        router: { pathname: "/" },
      }
    );

    wrapper.find("#user-menu").first().simulate("click");

    expect(mockOnClick).toBeCalledTimes(1);
  });
});
