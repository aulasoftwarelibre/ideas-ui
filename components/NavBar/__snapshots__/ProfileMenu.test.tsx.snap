// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu should be rendered closed with session 1`] = `
<DocumentFragment>
  <div
    class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
  >
    <div
      class="relative"
    >
      <div>
        <button
          aria-haspopup="true"
          aria-label="User menu"
          class="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
          id="user-menu"
        >
          <img
            class="h-8 w-8 rounded-full"
            src="/avatar.svg"
          />
        </button>
      </div>
      <div
        class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg"
      >
        <div
          aria-labelledby="user-menu"
          aria-orientation="vertical"
          class="py-1 rounded-md bg-white shadow-xs hidden"
          role="menu"
        >
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/user/profile"
            role="menuitem"
          >
            Editar perfil
          </a>
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/user/ideas"
            role="menuitem"
          >
            Actividades
          </a>
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/api/auth/signout"
            role="menuitem"
          >
            Salir
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProfileMenu should be rendered correctly without session 1`] = `<DocumentFragment />`;

exports[`ProfileMenu should be rendered opened with session 1`] = `
<DocumentFragment>
  <div
    class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
  >
    <div
      class="relative"
    >
      <div>
        <button
          aria-haspopup="true"
          aria-label="User menu"
          class="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
          id="user-menu"
        >
          <img
            class="h-8 w-8 rounded-full"
            src="/avatar.svg"
          />
        </button>
      </div>
      <div
        class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg"
      >
        <div
          aria-labelledby="user-menu"
          aria-orientation="vertical"
          class="py-1 rounded-md bg-white shadow-xs"
          role="menu"
        >
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/user/profile"
            role="menuitem"
          >
            Editar perfil
          </a>
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/user/ideas"
            role="menuitem"
          >
            Actividades
          </a>
          <a
            class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
            href="/api/auth/signout"
            role="menuitem"
          >
            Salir
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
